Assets/HoloToolkit/Utilities/Scripts/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.VR.WSA;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// StabilizationPlaneModifier handles the setting of the stabilization plane in several ways.
+    /// </summary>
+    public class StabilizationPlaneModifier : Singleton<StabilizationPlaneModifier>
+    {
+        [Tooltip("Checking enables SetFocusPointForFrame to set the stabilization plane.")]
+        public bool SetStabilizationPlane = true;
+        [Tooltip("Lerp speed when moving focus point closer.")]
+        public float LerpStabilizationPlanePowerCloser = 4.0f;
+        [Tooltip("Lerp speed when moving focus point farther away.")]
+        public float LerpStabilizationPlanePowerFarther = 7.0f;
+
+        [Tooltip("Used to temporarily override the location of the stabilization plane.")]
+        public Transform TargetOverride;
+
+        [Tooltip("Keeps track of position-based velocity for the target object.")]
+        public bool TrackVelocity = false;
+
+        [Tooltip("Use the GazeManager class to set the plane to the gazed upon hologram.")]
+        public bool UseGazeManager = true;
+
+        [Tooltip("Default distance to set plane if plane is gaze-locked.")]
+        public float DefaultPlaneDistance = 2.0f;
+
+        [Tooltip("Visualize the plane at runtime.")]
+        public bool DrawGizmos = false;
+
+        /// <summary>
+        /// Position of the plane in world space.
+        /// </summary>
+        private Vector3 planePosition;
+
+        /// <summary>
+        /// Current distance of the plane from the user's head. Only used when not using the target override
+        /// or the GazeManager to set the plane's position. 
+        /// </summary>
+        private float currentPlaneDistance = 4.0f;
+
+        /// <summary>
+        /// Tracks the previous position of the target override object. Used if velocity is being tracked.
+        /// </summary>
+        private Vector3 previousPosition;
+
+        /// <summary>
+        /// The framerate that the stabilization plane expects for velocity updates.
+        /// </summary>
+        private const float FOCUS_POINT_FRAMERATE = 60.0f;
```

Comment:
@NeerajW do you have a sense of our standard here? Does it differ from the private/public camelCase/PascalCase standard?